packages/flutter_tools/lib/src/android/android_device.dart

```diff
@@ -855,15 +854,24 @@ class _AndroidDevicePortForwarder extends DevicePortForwarder {
   }
 
   @override
-  Future<int> forward(int devicePort, { int hostPort }) async {
-    if ((hostPort == null) || (hostPort == 0)) {
-      // Auto select host port.
-      hostPort = await portScanner.findAvailablePort();
-    }
-
-    await runCheckedAsync(device.adbCommandForDevice(
+  Future<int> forward(int devicePort, {int hostPort}) async {
+    hostPort ??= 0;
+    final RunResult process = await runCheckedAsync(device.adbCommandForDevice(
       <String>['forward', 'tcp:$hostPort', 'tcp:$devicePort']
     ));
+    assert(process.exitCode == 0);
```

Comment:
Are we sure this will always succeed given correct user input?

Reply:
If not, then let's throw a `ProcessException` instead, where we can include the `stderr`, `exitCode`, etc.

Reply:
done.